torch/_dynamo/polyfills/__init__.py

```diff
@@ -318,6 +318,12 @@ def foreach_map_fn(*args):
     if not at_least_one_list:
         return op(*args[1:])
 
+    # Special handling for torch.mm
```

Comment:
So before this PR it was an open question whether this was needed or not, can you check what happens in your test without the special handling? It's possible this was already working, but it just needed a test.